### PR TITLE
Add command line confidence to shellIntegration API

### PR DESCRIPTION
**Pull request by Tyriar**
Part #145234

### CODE REVIEW

File: src/vs/workbench/api/common/extHostTerminalShellIntegration.ts [hunk 1]
```diff
@@ -160,30 +171,42 @@ class InternalTerminalShellIntegration extends Disposable {
 			get cwd(): URI | undefined {
 				return that._cwd;
 			},
-			executeCommand(commandLine): vscode.TerminalShellExecution {
-				that._onDidRequestShellExecution.fire(commandLine);
+			// executeCommand(commandLine: string): vscode.TerminalShellExecution;
+			// executeCommand(executable: string, args: string[]): vscode.TerminalShellExecution;
```
Comment (meganrogge) [inline src/vs/workbench/api/common/extHostTerminalShellIntegration.ts:175]: rm?

File: src/vs/workbench/api/common/extHostTypes.ts [hunk 2]
```diff
@@ -2001,6 +2001,12 @@ export enum TerminalExitReason {
 	Extension = 4
 }
 
+export enum TerminalShellExecutionCommandLineConfidence {
+	Low = 0,
```
Comment (meganrogge) [inline src/vs/workbench/api/common/extHostTypes.ts:2005]: could we have comments for these to explain what they mean?

File: src/vs/workbench/api/common/extHostTerminalShellIntegration.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/api/common/extHostTerminalShellIntegration.ts:175]: I left that in intentionally as those are the actual apis. I'd uncomment but TS doesn't like overloads

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh [hunk 3]
```diff
@@ -112,8 +112,8 @@ __vsc_update_cwd() {
 }
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh:112]: why were these moved to below?

File: src/vs/workbench/api/common/extHostTypes.ts (hunk 2 above)
Comment (Tyriar) [reply src/vs/workbench/api/common/extHostTypes.ts:2005]: These are documented in the actual API, I wanted to avoid duplicating that everywhere

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh (hunk 3 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh:112]: To get the command line before command execute fires, this should greatly improve the results and likely other features from shell integration too

File: src/vscode-dts/vscode.proposed.terminalShellIntegration.d.ts [hunk 4]
```diff
@@ -58,9 +41,67 @@ declare module 'vscode' {
...
+
+		/**
+		 * The command line value confidence is medium. This means that the value was read from the
+		 * terminal buffer using markers reported by the shell integration script. The command is
+		 * single-line and does not start on the very left-most column (which is unusual).
+		 */
+		Medium = 1,
+
+		/**
+		 * The command line value confidence is high. This means that the value was explicitly send
```
Comment (meganrogge) [inline src/vscode-dts/vscode.proposed.terminalShellIntegration.d.ts:99]:
```suggestion
		 * The command line value confidence is high. This means that the value was explicitly sent
```

**Review by meganrogge — DISMISSED** [review]

**Review by meganrogge — APPROVED** [review]
